### ISSUE TITLE
Enumerate the possible `jsdomError.type` strings

### ISSUE THREAD

**Issue by joeframbach** [jsdom/jsdom#3596]
### Basic info:

- **Node.js version:** N/A
- **jsdom version:** latest

### Minimal reproduction case

```js
const { VirtualConsole } = require("jsdom");

const virtualConsole = new VirtualConsole();

virtualConsole.on("jsdomError", error => {
  if (error.type === "this error type doesn't exist") {
    console.log("Typescript should warn me here!");
  }
});
```

I filed a PR with `@types/jsdom` at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66630. I would like the jsdom team to take a look and discuss how we can better enumerate the jsdomError types. In my PR I note "There is no single enumeration of error type strings. It is not clear if jsdom considers this part of a stable API surface." Please give commentary on this. Are the strings going to be "stable" within a non-major version bump? It's not part of the API surface and not defined in DefinitelyTyped, so it is not guaranteed.

**Comment by domenic** [comment]
Since they are not documented, no, I would not consider them stable API surface. We've had ambitions of maybe documenting them, but maybe also improving them before doing so, so... *shrug*.

**Comment by joeframbach** [comment]
Perhaps this is a conversation for DefinitelyTyped, then. Are the types in DefinitelyTyped meant to imply the intention behind the source package, or is it supposed to one-for-one mirror the types used internally in the source package? Should I continue with my PR in DefinitelyTyped since it describes how jsdom is written, or should I not, because jsdom doesn't actually define that type?

**Comment by domenic** [comment]
Yeah, we have no association with the DefinitelyTyped project, so I don't know how to answer those questions.

**Comment by domenic** [comment]
Let me repurpose this issue into one whose goal is:

- Enumerate the existing types
- Decide which we want to keep, or revise
- Document them all
- Add a new feature that will allow filtering out which jsdomErrors are reported. (This helps with common complaints that people don't care about CSS errors.)
